agent/config/config.go

```diff
@@ -149,6 +157,169 @@ type Nomad struct {
 	SkipVerify bool `hcl:"skip_verify,optional"`
 }
 
+// Consul contains the configuration information necessary to
+// communicate with a Consul server.
+type Consul struct {
+	// Addr is the HTTP endpoint address of the local Consul agent
+	//
+	// Uses Consul's default and env var.
+	Addr string `hcl:"address"`
+
+	// TimeoutHCL is used by Consul HTTP Client
+	TimeoutHCL string        `hcl:"timeout" json:"-"`
+
+	// Token is used to provide a per-request ACL token. This options overrides
+	// the agent's default token
+	Token string `hcl:"token"`
+
+	// Auth is the information to use for http access to Consul agent
+	Auth string `hcl:"auth"`
+
+	// EnableSSL sets the transport scheme to talk to the Consul agent as https
+	//
+	// Uses Consul's default and env var.
+	EnableSSL *bool `hcl:"ssl"`
+
+	// VerifySSL enables or disables SSL verification when the transport scheme
+	// for the consul api client is https
+	//
+	// Uses Consul's default and env var.
+	VerifySSL *bool `hcl:"verify_ssl"`
+
+	// CAFile is the path to the ca certificate used for Consul communication.
+	//
+	// Uses Consul's default and env var.
+	CAFile string `hcl:"ca_file"`
+
+	// CertFile is the path to the certificate for Consul communication
+	CertFile string `hcl:"cert_file"`
+
+	// KeyFile is the path to the private key for Consul communication
+	KeyFile string `hcl:"key_file"`
+
+	// Namespace sets the Consul namespace used for all calls against the
+	// Consul API. If this is unset, then we don't specify a consul namespace.
+	Namespace  string `hcl:"namespace"`
+
+	// Datacenter sets the Consul datacenter used for all calls against the
+	// Consul API. If this is unset, then we don't specify a consul datacenter.
+	Datacenter string `hcl:"datacenter"`
+}
+
+func (c *Consul) MergeWithDefault() (*consulapi.Config, error) {
+	if c == nil {
+		return nil, nil
+	}
+
+	cfg := consulapi.DefaultConfig()
+
+	if c.TimeoutHCL != "" {
+	}
```

Comment:
Does this need to be removed? I notice the same conditional is below (L225) and parses the duration.

Reply:
oops, leftover copy-pasta 🤦